docs/cow-protocol/concepts/order-types/pay-debt-flash-loans.md

```diff
@@ -0,0 +1,104 @@
+---
+sidebar_position: 6
+---
+
+# Repay debt with collateral using flash loans
+
+A key use case of flash loans is the ability to repay debt with collateral, since flash loans allow users to close or reduce their debt positions without needing upfront liquidity.
+
+Instead of requiring users to have assets readily available to repay their loans, flash loans enable them to temporarily borrow the needed funds, use those funds to repay their debt, and immediately reclaim their locked collateral. Once the collateral is received, it can be swapped or liquidated to cover the borrowed amount, ensuring the loan is repaid within the same transaction.
+
+This approach eliminates the need for users to preemptively sell assets or find external liquidity, making it a highly flexible and efficient way to manage debt positions.
+
+### How to take out a flash loan
+
+This can be achieved using a buy order where:
+
+- The **buy token** is the flash loaned asset.
+- The **buy amount** is the exact amount needed to repay the debt (it must be sufficient to complete the transaction; otherwise, the transaction will revert!).
+- The **sell token** is the asset used as collateral.
+- The **sell amount** is the full collateral amount.
+
+The receiver must always be the settlement contract, while the protocol ensures that the funds are properly directed to the appropriate address.
+
+When it comes to repaying debt, the approach depends on the lender's contract terms. Some lenders allow a third party to pay off the debt and transfer the unlocked funds to the owner's account, while others may not.
+
+#### Lender allows a third-party to repay your debt
+
+In this case, the user can sign a pre-hook that deploys a [cowshed](../../reference/sdks/cow-sdk/classes/CowShedHooks.md) to repay the debt using flash-loaned tokens. The underlying user order is then solely for repaying the required flash loan. Since the repayment is handled via `cowshed`, the user can be either an EOA (Externally Owned Account) or a contract (e.g., a SAFE wallet).
+
+#### Lender does not allow a third-party to repay your debt
+
+If the lender does not allow third-party debt repayment, the process becomes more involved. In this case, the repayment must come directly from the owner's account, meaning the owner cannot simply be an EOA (Externally Owned Account). Instead, it must be a smart contract (e.g.,a SAFE wallet), which can execute more complex transactions atomically. This is important because a contract can facilitate operations such as selling assets and repaying the debt in a single transaction, something an EOA cannot do.
```

Comment:
> Instead, it must be a smart contract (e.g.,a SAFE wallet), which can execute more complex transactions atomically. This is important because a contract can facilitate operations such as selling assets and repaying the debt in a single transaction, something an EOA cannot do.

The reason why an SC can work in this case is that it can be written in a way that let's a third party initiate the withdrawal of collateral tokens. This is not possible when the collateral owner is an EOA.